Apsitvarkom.UnitTests/DataAccess/PollutedLocationDTODatabaseRepositoryTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿using Apsitvarkom.DataAccess;
+using Apsitvarkom.Models;
+using AutoMapper;
+using Apsitvarkom.Models.Mapping;
+using Moq;
+using MockQueryable.Moq;
+using System.Globalization;
+using static Apsitvarkom.Models.Enumerations;
+
+namespace Apsitvarkom.UnitTests.DataAccess;
+
+public class PollutedLocationDTODatabaseRepositoryTests
+{
+    private IMapper m_mapper = null!;
+    private Mock<IPollutedLocationContext> m_mockContext = null!;
+
+    [SetUp]
+    public void SetUp()
+    {
+        var config = new MapperConfiguration(cfg =>
+        {
+            cfg.AddProfile<PollutedLocationProfile>();
+        });
+        config.AssertConfigurationIsValid();
+        m_mapper = config.CreateMapper(); 
+        m_mockContext = new Mock<IPollutedLocationContext>();
+    }
+
+    #region Constructor tests
+    [Test]
+    public void PollutedLocationDTODatabaseRepositoryConstructor_HappyPath() =>
+        Assert.DoesNotThrow(() => new PollutedLocationDTODatabaseRepository(m_mockContext.Object, m_mapper));
+    #endregion
+
+    #region GetAllAsync tests
+    [Test]
+    public async Task GetAllAsync_DbContextHasNoInstances_EmptyListReturned()
+    {
+        var dbInstances = new List<PollutedLocation>();
+        var mock = dbInstances.AsQueryable().BuildMockDbSet();
+        m_mockContext.Setup(m => m.PollutedLocations).Returns(mock.Object);
+        var dataManager = new PollutedLocationDTODatabaseRepository(m_mockContext.Object, m_mapper);
+
+        var instances = (await dataManager.GetAllAsync()).ToArray();
+
+        Assert.That(instances, Is.Empty);
+    }
+
+    [Test]
+    public async Task GetAllAsync_DbContextHasOneInstance_SingleInstanceReturnedWithCorrectProperties()
+    {
+        var id = Guid.NewGuid();
+        var notes = "notes";
+        var spotted = new DateTime(2022, 11, 12, 19, 23, 30);
+        var coordinates = new Coordinates { Latitude = 47.12, Longitude = -41.1251 };
+        var radius = 35;
+        var progress = 67;
+        var severity = LocationSeverityLevel.Moderate;
+        var dbInstances = new List<PollutedLocation>
+        {
+            new() { Id = id, Coordinates = coordinates, Radius = radius, Severity = severity, Spotted = spotted, Notes = notes, Progress = progress }
+        };
```

Comment:
Just wondering why is everything a separate variable? 😄 and then in line 61 you just have a lot of repeated variable names. I would just create the object inline (inside the `new()` in `new List<PollutedLocation>{}`)

Reply:
Or just create a single `new PollutedLocation` and add it to the `dbInstances` list (I just saw that you're asserting it later)